### ISSUE TITLE
Page scroll is not prevented when mouseScrolled is used

### ISSUE THREAD

**Issue by nbeloglazov** [processing-js/processing-js#220]
When `mouseScrolled` is used to handle mouse scroll event page still scrolls. It's fine when page doesn't have scrollbars but when it does - it looks pretty weird. Example: https://jsfiddle.net/u7hphbjq/19/ 

Scroll up - circle size increases. Scroll down - circle size decreases, but page scrolls down as well, which is undesirable. 

Another issue is that scroll handler is called when mouse is over any element on a page, not only canvas. I think it might be better if by default scroll handler is called only when mouse is over canvas and provide an option to have global scroll event (which might be triggered by [globalKeyEvents](http://processingjs.org/reference/globalKeyEvents/) for example). This way it will be aligned with how other keyboard and mouse handlers work. 

I think it can be achieved by adding `e.preventDefault()` to [mouseWheelHandler](https://github.com/processing-js/processing-js/blob/master/src/P5Functions/touchmouse.js#L302) when user provided a handler. And also [register](https://github.com/processing-js/processing-js/blob/master/src/P5Functions/touchmouse.js#L323) event listener on canvas element instead of document.

**Comment by Pomax** [comment]
good points; scroll should indeed be handled by the canvas exclusively rather than the document, and not bubble.

**Comment by MatthiasSaihttam** [comment]
I believe I've found an fix for this issue, namely adding 
`e.preventDefault();
return false;`
after [this](https://github.com/processing-js/processing-js/blob/11fbef817c6a5f48765cbc77238e928e5264f49d/processing.js#L7803) line. I would go ahead and create a PR, but I'm not very familiar with this repository, and in particular don't know how to update the minimized version of the source. If you could provide some assistance, I would appreciative.

**Comment by GoToLoop** [comment]
"processing.js" is the generated compiled file. It's useless to modify that 1.
Instead we need to find out within folder "src/" which file we're interested in modifying.

Seems like it's located within "touchmouse.js":
https://GitHub.com/processing-js/processing-js/blob/master/src/P5Functions/touchmouse.js

It's enough to append a `return false;` at the end of **mouseWheelHandler()** callback handler:

```js

  p.mouseScroll = delta;

  if (delta && typeof p.mouseScrolled === 'function') {
    p.mouseScrolled();
  }

  return false; // workaround fix here!
};
```

**Comment by Pomax** [comment]
we'll still might need the `preventDefault()` on the event as well, as newer browsers no longer treat a `false` return value as a propagation control, but that *should* do it.

If you file a PR for the change in the file @GoToLoop pointed at, then an `npm test` to verify it didn't break anything will take care of the build process for you. We might need to write a custom test with some dummy paragraphs of text that runs for several page-heighs and a canvas element with a sketch that relies on mouseScrolled, so that we have a test for verifying to failed before, and works with  your fix.

**Comment by GoToLoop** [comment]
> ... as newer browsers no longer treat a false return value as a propagation control, ...

Can you mention such a browser?! `return false;` is what p5.js recommends btW:
http://p5js.org/reference/#/p5/mouseWheel

```js
function mouseWheel(event) {
  print(event.delta);
  //move the square according to the vertical scroll amount
  pos += event.delta;
  //uncomment to block page scrolling
  //return false;
}
```

**Comment by MatthiasSaihttam** [comment]
Ok, I created a PR. I choose to use both `e.preventDefault()` and `e.stopPropagation()`, which, from what I read, does the same thing as `return false` but is more compatible with the current standards. I also choose to put that code inside the if statement, so it only takes effect if the code has a `mouseScrolled` function. I prefer it that way, but if you've a reason to make the behavior unconditional I'd willing to make the code change. Hope I made the changes correctly and this can get merged.

**Comment by GoToLoop** [comment]
> ... but is more compatible with the current standards.

Although they don't work w/ older IE versions... But who cares, huh?!  :P

**Comment by GoToLoop** [comment]
> I also choose to put that code inside the if statement, so it only takes effect if the code has a **mouseScrolled()** function.

Perhaps you're right about that. Let it propagate if we didn't provided our own handler callback.

**Comment by MatthiasSaihttam** [comment]
This issue is fixed with #253 being merged, and can be closed!
